added episode.py

Egge7 opened the pull request:
I added episode.py that includes the command "/episode".
"/episode" crawls einundzwanzig.space for the latest podcast episodes and returns the URL. The podcast format can be specified by adding following arguments: "alle", "interview", "lesestunde", "news" or "weg". If no arguments are used, "/episode" will return the lastest episode of all format.

PatrickLemke commented on src/bot.py:
```diff
@@ -104,6 +111,7 @@ def run(bot_token: str):
     sat_in_usd_handler = CommandHandler('satinusd', sat_in_usd_command, run_async=True)
     blockzeit_handler = CommandHandler('blockzeit', blockzeit_command, run_async=True)
     moskauzeit_handler = CommandHandler('moskauzeit', moskauzeit_command, run_async=True)
+    episode_handler = CommandHandler('episode', episode_command, run_asynch=True)
```
```suggestion
    episode_handler = CommandHandler('episode', episode_command, run_async=True)
```

PatrickLemke commented on src/episode.py:
```diff
@@ -0,0 +1,36 @@
+urlInterviews = 'https://einundzwanzig.space/podcast/interviews/'
+
+def getEpisode(url):
+    r = requests.get(url)
```
You should add a timeout, see https://docs.python-requests.org/en/master/user/quickstart/#timeouts

PatrickLemke commented on src/episode.py:
```diff
@@ -0,0 +1,36 @@
+from telegram.ext import Updater
+import requests
+from bs4 import BeautifulSoup
```
We need to add this to the requirements.txt, otherwise the dependency will not be installed

PatrickLemke commented on src/episode.py:
```diff
@@ -0,0 +1,36 @@
+from bs4 import BeautifulSoup
+from telegram.ext import CommandHandler
+
+# Define podcast formats
```
I would use a separate variable for the url and then just add the path afterwards. Probably best to do it in the config file.

PatrickLemke commented on src/episode.py:
```diff
@@ -0,0 +1,36 @@
+urlAll = 'https://einundzwanzig.space/podcast/'
+urlInterviews = 'https://einundzwanzig.space/podcast/interviews/'
+
+def getEpisode(url):
```
I think this function needs some error handling. What happens if the request fails? Should we inform the user?

PatrickLemke reviewed (commented):
I think this needs some more testing. Did you try this with your own bot?

Egge7 commented:
- added timeout and error handling for requests.get. If the request fails for some reason, users will get notified and receive a link to EINUNDZWANZIG Spotify instead.
- removed root URL from variables and moved it to config file
- fixed typo in bot.py (synch/sync)
- updated requirements.txt

I did test the original commit with my own bot, but in a "single file" environment with out Docker etc.

PatrickLemke commented:
I've added two more commits with some style changes. Let me know what you think.

PatrickLemke commented on src/episode.py:
```diff
@@ -0,0 +1,48 @@
+        format = "alle"
+    
+    if format == "news":
+        message = getEpisode(config.EINUNDZWANZIG_URL + urlNews)
```
I think it makes more sense to just provide the path of the url as the argument here, since the EINUNDZWANZIG_URL is always the same. It should just be added in the getEpisode function to avoid duplication.

Egge7 commented:
Thanks for cleaning up! f-strings and using variables in strings were new concepts for me (as you might have noticed...). I also removed the root URL from the argument passed to the function and included it in getEpisode.
